com.unity.netcode.gameobjects/Tests/Runtime/NetworkVariableTests.cs

```diff
@@ -4,8 +4,76 @@
 using UnityEngine.TestTools;
 using NUnit.Framework;
 
+
+
 namespace Unity.Netcode.RuntimeTests
 {
+    public unsafe struct FixedStringStruct : INetworkSerializable
+    {
+        private const int k_MaxSize = 128;
+        public fixed char FixedString[k_MaxSize];
+        public int StringSize;
+
+        public void SetIt(string valueToSet)
+        {
+            Assert.False(valueToSet.Length >= k_MaxSize);
+            fixed(char* charPtr = FixedString)
+            {
+                for(int i = 0; i < valueToSet.Length; i++)
+                {
+                    charPtr[i] = valueToSet[i];
+                }
+            }
+            StringSize = valueToSet.Length;
+        }
+
+        public string GetIt()
+        {
+            var retValue = string.Empty;
+            fixed (char* charPtr = FixedString)
+            {
+                for (int i = 0; i < StringSize; i++)
+                {
+                    retValue += FixedString[i];
+                }
+            }
+            return retValue;
+        }
+
+        public void NetworkSerialize(NetworkSerializer serializer)
+        {
+            if (serializer.IsReading)
```

Comment:
I want to come back and revisit this, just not today.  I also would like to re-write with `FixedString32` @andrews-unity , and I got it close but again, would like to get this in and improve later.